File: Tutorial1ProvisionMethod/src/main/java/com/example/tutorial1daggerbasics2/di/VehicleModule.java
package com.example.tutorial1daggerbasics2.di;

import com.example.tutorial1daggerbasics2.model.Motor;
import com.example.tutorial1daggerbasics2.model.Vehicle;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class VehicleModule {

    @Singleton
    @Provides
    Motor provideMotor(){
        return new Motor();
    }

    @Singleton
    @Provides
    Vehicle provideVehicle(){
        return new Vehicle(new Motor());
    }
}

File: Tutorial1-3ProvisionMethod/src/main/java/com/example/tutorial1daggerbasics2/di/VehicleModule.kt
package com.example.tutorial1daggerbasics2.di

import com.example.tutorial1daggerbasics2.model.Motor
import com.example.tutorial1daggerbasics2.model.Vehicle
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class VehicleModule {

    @Singleton
    @Provides
    fun provideMotor(): Motor {
        return Motor()
    }

    @Singleton
    @Provides
    fun provideVehicle(): Vehicle {
        return Vehicle(Motor())
    }
}